packages/workflow/src/internals.ts

```diff
@@ -617,22 +618,21 @@ export class Activator implements ActivationHandler {
         throw new IllegalStateError(
           `Unsupported internal patch number: ${internalPatchNumber} > ${LATEST_INTERNAL_PATCH_NUMBER}`
         );
-      if (this.internalPatchNumber < internalPatchNumber) this.internalPatchNumber = internalPatchNumber;
+      if (this.knownInternalPatchNumber < internalPatchNumber) this.knownInternalPatchNumber = internalPatchNumber;
     } else {
       this.knownPresentPatches.add(activation.patchId);
     }
   }
 
   public checkInternalPatchAtLeast(minimumPatchNumber: number): boolean {
-    if (this.internalPatchNumber >= minimumPatchNumber) return true;
-    if (!this.info.unsafe.isReplaying) {
-      this.internalPatchNumber = minimumPatchNumber;
+    const usePatch = !this.info.unsafe.isReplaying || this.knownInternalPatchNumber >= minimumPatchNumber;
+    if (usePatch && this.sentInternalPatchNumber < minimumPatchNumber) {
       this.pushCommand({
-        setPatchMarker: { patchId: `__sdk_internal_patch_number:${LATEST_INTERNAL_PATCH_NUMBER}`, deprecated: false },
+        setPatchMarker: { patchId: `__sdk_internal_patch_number:${minimumPatchNumber}`, deprecated: false },
```

Comment:
Hmm.. this will cause us to emit a patch for every patch increment.
That's probably fine considering that we won't use this mechanism anymore.

I'd strongly consider scrapping all of this complexity and using a single hard-coded `__sdk_internal_patch_number:1`.

Reply:
Yeah, let's just do that and implement the next patch using the new dedicated mechanism.